w3-aggregation.md

```diff
@@ -0,0 +1,337 @@
+
+A Storefront facilitates data storage services to applications and users, getting the requested data stored into Filecoin deals asynchronously.
+
+### Authority
+
+_Authority_ is a [principal] that executes invoked capabilities.
+
+### Verifier
+
+A Component of the [authority] that performs UCAN validation
+
+# Protocol
+
+## Overview
+
+A Storefront is the entry point for user/application data into web3. It will act on behalf of users to move data around into different storage points. One of the key storage presences may be Filecoin Storage Providers.
+
+### Authorization
+
+Broker MUST have an authorization mechanism for allowed Storefront principals (e.g. web3.storage). Either by out-of-bound exchange of information or through a well defined API. In other words, broker can authorize invocations from `did:web:web3.storage` by validating signature from did. This way, it allows web3.storage to rotate keys without having too coordinate that with the broker.
+
+### Storefront offers broker an aggregate
+
+When a Storefront has enough content to fulfill an aggregate (each broker MAY have different requirements), a Filecoin deal for an aggregate SHALL be requested by a `aggregate/offer` invocations. Deal negotiations with Filecoin Storage Providers will be handled out off band. Broker SHOULD generate a receipt to acknowledge the received request.
+
+```mermaid
+sequenceDiagram
+    participant Storefront as 🌐<br/><br/>did:web:web3.storage
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+
+    Storefront->>Authority: invoke `aggregate/offer`
+    Note left of Authority: Request offer to be queued
+```
+
+### Broker queues the offer
+
+Once broker successfully gets an offer, the offer gets queued for review. A receipt is created to proof the transition of `aggregate/offer` state from `null` into `queued`.
+
+This receipt MUST contain a follow up task in the (`.fx.join` field) that is run when submitted request is processed. It MAY succeed (if aggregate was accepted) or fail (if aggregated was  determined to be invalid). The result of the subsequent task CAN be looked up using its receipt.
+
+```mermaid
+sequenceDiagram
+    participant Authority as 🌐<br/><br/>did:web:spade.storage
+    participant Storefront as 🌐<br/><br/>did:web:web3.storage
+
+    Storefront->>Authority: invoke `aggregate/offer`
+    Note left of Storefront: Queueing offer
+    Authority-->>Storefront: receipt issued
+```
+
+### Broker reviews and handles the offer
+
+When a broker pops the offer from the queue, the offer details MUST be retrievable. With the offer details, the broker MAY interact with available Filecoin Storage Providers, in order to establish a previously determined number of deals. Depending on storage providers availability, as well as the content present in the offer, the aggregate MAY be handled or not. A receipt is created to proof the transition of `aggregate/offer` state from `queued` into `accepted` or `denied`.
```

Comment:
> When a broker pops the offer from the queue, the offer details MUST be retrievable.

"pops" suggests a FILO structure, i.e. not a queue, a stack.

I'm not sure I understand what this means - might need re-wording.

Is the offer status not availabile for querying until it reaches the front of the queue? Oh, you mean the broker should be able to retrieve the details of the aggregate from us?

> MAY interact with available Filecoin Storage Providers

Why is this may? It _will_ interact with SPs to facilitate a deal.

Reply:
Reworded in https://github.com/web3-storage/specs/pull/51/commits/91e928080484202966556f87d020404fb31035fe

> Oh, you mean the broker should be able to retrieve the details of the aggregate from us?

yes!